allennlp/training/optimizers.py

```diff
@@ -4,14 +4,6 @@
 
 from allennlp.common.params import Params  # pylint: disable=unused-import
 
-optimizers = {  # pylint: disable=invalid-name
-        "adam": torch.optim.Adam,
-        "adagrad": torch.optim.Adagrad,
-        "adadelta": torch.optim.Adadelta,
-        "sgd": torch.optim.SGD,
-        "rmsprop": torch.optim.RMSprop,
-}
-
 
 def get_optimizer_from_params(model_parameters: List[torch.Tensor],
```

Comment:
I think it'd be more consistent with the rest of the library to just have this be a class:

```python
class Optimizer:
    def from_params(model_parameters: List[torch.Tensor], params: Union[Params, str]) -> torch.optim.Optimizer:
        ...
```